.azure-pipelines.yml

```diff
@@ -0,0 +1,124 @@
+name: $(Build.SourceBranch)-$(date:yyyy-MM-dd)$(rev:.r)
+trigger:
+  branches:
+    include:
+    - 'master'
+  tags:
+    include:
+    - '*'
+    exclude:
+      - 'dev'
+
+jobs:
+- job: rustfmt
+  pool:
+    vmImage: 'macos-10.14'
+  steps:
+  - checkout: self
+    submodules: true
+  - template: ci/azure-install-rust.yml
+  - script: rustup component add rustfmt
+    displayName: Add rustfmt
+  - script: cargo fmt --all -- --check
+    displayName: Check formatting
+  variables:
+    toolchain: stable
+
+- job: Test
+  strategy:
+    matrix:
+      windows-stable:
+        imageName: 'vs2017-win2016'
+        toolchain: stable
+      linux-stable:
+        imageName: 'ubuntu-16.04'
+        toolchain: stable
+      mac-stable:
+        imageName: 'macos-10.14'
+        toolchain: stable
+      mac-beta:
+        imageName: 'macos-10.14'
+        toolchain: beta
+      mac-nightly:
+        imageName: 'macos-10.14'
+        toolchain: nightly
+
+  pool:
+    vmImage: $(imageName)
+
+  steps:
+    - checkout: self
+      submodules: true
+    - template: ci/azure-install-rust.yml
+
+    - script: cargo fetch
+      workingDirectory: tools/witx
+      displayName: Fetch cargo dependencies
+
+    - script: cargo build
+      workingDirectory: tools/witx
+      displayName: Cargo build
+
+    - bash: cargo test --all
+      workingDirectory: tools/witx
+      displayName: Cargo test
+      env:
+        RUST_BACKTRACE: 1
+
+- job: Build
```

Comment:
Most of the goop here (weird env vars, centos, libstdc++, etc) is only related to the binary releases of the `wasmtime` binary and associated tools. Is it intended though that the witx tool is available as a binary through this repo? If not this can probably all largely be dleted, if so though we can probably simplify the Linux build by quite a lot since `witx` doesn't depend on C++ (unlike wasmtime)

Reply:
The `witx` executable is useful to distribute, since it validates a `.witx` document.

And yeah, lets drop all the C++ stuff as well. I didn't actually know what all of that was there for.

Reply:
Ok cool, in that case if we're building releases then I think we'll want to switch the Linux release to using the `x86_64-unknown-linux-musl` target to statically link everything. That way we don't have to deal with centos and can use a modern build environment. The Windows/OSX bits would need to stay, but this isn't likely to be a super widely used executable anyway so none of the weird build config is honestly all that important to keep.

Reply:
Yeah, I would hope we can avoid all that docker + sudo + static linking complexity.  Building against a standard debian/stable (of ubuntu LTS perhaps) image should be good enough to produce a widely runnable binary.